### PR TITLE
update and delete post

### PR DESCRIPTION
**Pull request by duynaregit-technology**

### CODE REVIEW

File: app/Http/Kernel.php [hunk 1]
```diff
@@ -66,5 +66,7 @@ class Kernel extends HttpKernel
         'verified' => \Illuminate\Auth\Middleware\EnsureEmailIsVerified::class,
         'checkRoleLogin' => \App\Http\Middleware\CheckRoleLogin::class,
         'checkVerify' => \App\Http\Middleware\CheckVerify::class,
+        'checkUserHasPermissions' => \App\Http\Middleware\CheckUserHasPermissions::class,
```
Comment (TruongPM-Regit) [inline app/Http/Kernel.php:69]: https://laravel.com/docs/11.x/authorization

File: app/Http/Kernel.php [hunk 2]
```diff
@@ -66,5 +66,7 @@ class Kernel extends HttpKernel
...
+        'shareUserData' => \App\Http\Middleware\ShareUserData::class,
```
Comment (TruongPM-Regit) [inline app/Http/Kernel.php:70]: [view compose](https://laravel.com/docs/11.x/views#sharing-data-with-all-views)

File: app/Services/User/PostService.php [hunk 3]
```diff
@@ -36,4 +38,77 @@ public function create($data): void
...
+    {
+        try {
+            return Post::with('categories')->findOrFail($id);
+        } catch (Exception $e) {
+            Log::error($e);
+            throw $e;
+        }
+    }
+
+    public function getListPostByUserId(int $user_id, int $id): Collection
```
Comment (TruongPM-Regit) [inline app/Services/User/PostService.php:52]: sai convention

File: resources/js/user/post.js [hunk 4]
```diff
@@ -37,4 +37,27 @@ $(document).ready(function () {
...
+            $(this).text(shortContent + '...');
+        }
+    });
+
+    $('#showMenu').on('click', function () {
+        $('#menu').toggle();
+    });
+
+    $('.ticked').on('click', function () {
+        const postId = $(this).data('post').id;
```
Comment (namtpregit) [inline resources/js/user/post.js:54]: const { id: postId } = $(this).data('post')

File: app/Services/User/PostService.php [hunk 5]
```diff
@@ -36,4 +38,77 @@ public function create($data): void
...
+    {
+        try {
+            return Post::public()
+                ->with('categories')
+                ->where('user_id', $user_id)
+                ->where('id', '<>', $id)
+                ->withCount('likes', 'comments')
+                ->orderBy('likes_count', 'desc')
+                ->orderBy('comments_count', 'desc')
+                ->limit(4)
```
Comment (TruongPM-Regit) [inline app/Services/User/PostService.php:62]: constant

File: app/Services/User/PostService.php [hunk 6]
```diff
@@ -36,4 +38,77 @@ public function create($data): void
...
+
+    public function getListPostByUserId(int $user_id, int $id): Collection
+    {
+        try {
+            return Post::public()
+                ->with('categories')
+                ->where('user_id', $user_id)
+                ->where('id', '<>', $id)
+                ->withCount('likes', 'comments')
+                ->orderBy('likes_count', 'desc')
```
Comment (TruongPM-Regit) [inline app/Services/User/PostService.php:60]: k random order à

File: app/Services/User/PostService.php [hunk 7]
```diff
@@ -36,4 +38,77 @@ public function create($data): void
...
+        try {
+            $post = Post::findOrFail($id);
+            if (isset($data['image'])) {
+                $fileName = time() . $data['image']->getClientOriginalName();
+                $imagePath = config('config.dir_image') . $data['image'];
+                if (!Storage::exists($imagePath)) {
+                    Storage::delete(config('config.dir_image') . $post->image);
+                }
+                $data['image']->storeAs(config('config.dir_image'), $fileName);
+            } else {
```
Comment (TruongPM-Regit) [inline app/Services/User/PostService.php:82]: sao phải else

File: app/Services/User/PostService.php [hunk 8]
```diff
@@ -36,4 +38,77 @@ public function create($data): void
...
+                $fileName = $post->image;
+            }
+            $post->update([
+                'content' => $data['content'],
+                'image' => $fileName,
```
Comment (TruongPM-Regit) [inline app/Services/User/PostService.php:87]: .

File: routes/web.php [hunk 9]
```diff
@@ -66,8 +66,14 @@ function () {
...
     Route::get('/', [UserHomeController::class, 'home'])->name('home');
-    Route::prefix('post')->name('post.')->middleware('auth')->group(function () {
-        Route::get('/create', [UserPostController::class, 'create'])->name('create');
-        Route::post('/store', [UserPostController::class, 'store'])->name('store');
+    Route::prefix('post')->name('post.')->group(function () {
+        Route::group(['middleware' => 'auth'], function () {
+            Route::get('/create', [UserPostController::class, 'create'])->name('create');
+            Route::post('/store', [UserPostController::class, 'store'])->name('store');
+            Route::get('/edit/{id}', [UserPostController::class, 'edit'])->name('edit');
+            Route::post('/update/{id}', [UserPostController::class, 'update'])->name('update');
```
Comment (TruongPM-Regit) [inline routes/web.php:74]: put

**Pull request merged**